dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigProviderResolver.java

```diff
@@ -157,7 +153,7 @@ private static String getApplicationName() {
         if (applicationName == null) {
             //There are cases where the Config is used by a global component or we just can't work out which app it is. Then we fall back to this global name.
             //Configs used "globally" will be shutdown when the server is shutdown
-            applicationName = GLOBAL_CONFIG_APPLICATION_NAME;
+            applicationName = ConfigWrapper.GLOBAL_CONFIG_APPLICATION_NAME;
         }
```

Comment:
Can we add some trace here to identify when and why this happens?

I guess it's a bit tricky, on the one had we want to dump the stack to see who called us, on the other hand there are valid cases where there's no application and in those cases we don't want the cost of dumping the stack.

Maybe we could just include the thread name (which would show if it was an unmanaged thread) and the TCCL?

Reply:
i'll try to add some debug before we merge this ... will wait for Brian to try out the TCK first though